src/screens/login/LoginScreen.tsx

```diff
@@ -0,0 +1,144 @@
+   * Logs in a user from the data in the fields after it passes the validation from react hook forms.
+   * After successful or unsuccessful login, resets the text fields to be empty and persists the login.
+   *
+   * @param data object containing fields from each input wrapped by a controller
+   */
+  const onSubmit = (data: any) => {
+    console.log('handling login', data);
+    handleLogin(data.email, data.password).then(() => {
+      reset({
+        email: '',
+        password: '',
+      });
+    });
+  };
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+      <SafeAreaView style={LoginScreenStyles.container}>
+        <View style={LoginScreenStyles.logoDot}></View>
+        <View style={LoginScreenStyles.logoBody}></View>
+        <Text style={LoginScreenStyles.h1}>
+          Hey there<Text style={LoginScreenStyles.blueText}>.</Text>
+        </Text>
+        <Text style={LoginScreenStyles.h2}>We're glad you're back.</Text>
+        <View style={LoginScreenStyles.emailContainer}>
+          <Controller
+            control={control}
+            render={({ field: { onChange, onBlur, value } }) => (
+              <TextInput
+                style={LoginScreenStyles.emailInput}
+                onBlur={onBlur}
+                onChangeText={value => onChange(value)}
+                value={value}
+                placeholder='Email'
+              />
+            )}
+            name='email'
+            rules={{ required: true }}
+          />
+          <View style={LoginScreenStyles.passwordContainer}>
+            <Controller
+              control={control}
+              render={({ field: { onChange, onBlur, value } }) => (
+                <TextInput
+                  style={LoginScreenStyles.passwordInput}
+                  onBlur={onBlur}
+                  onChangeText={value => onChange(value)}
+                  value={value}
+                  placeholder='Password'
+                  secureTextEntry={isPasswordHidden}
+                />
+              )}
+              name='password'
+              rules={{ required: true }}
+            />
+            <Pressable
+              style={LoginScreenStyles.viewPasswordButton}
+              onPress={() => setIsPasswordHidden(!isPasswordHidden)}>
+              {isPasswordHidden ? (
+                <Entypo
+                  name='eye-with-line'
+                  size={24}
+                  color={COLORS.NIGHTLIGHT_GRAY}
+                />
+              ) : (
+                <Entypo name='eye' size={24} color={COLORS.NIGHTLIGHT_GRAY} />
+              )}
+            </Pressable>
+          </View>
+
+          <View style={LoginScreenStyles.forgotPasswordContainer}>
+            <Text style={LoginScreenStyles.forgotPassword}>
+              Forgot password?
+            </Text>
+          </View>
+          <Pressable
+            style={LoginScreenStyles.signInButton}
+            onPress={handleSubmit(onSubmit)}>
```

Comment:
```suggestion
            onPress={() => handleSubmit(onSubmit)}>
```
not sure abt this? does this behave differently with hook forms?